### PR TITLE
[MDO] Fix empty subject by sorting on the first to last  message and get first one (oldest should be the original)

### PR DESCRIPTION
**Pull request by iserrano76**
**Issue:**
If you have more that one copy in message trace not always shows subject

**Reason:**
With the current filter we do not take always the original one, the oldest one.

**Fix:**
Fix empty subject by sorting on the first to last  message and get first one (oldest should be the original and has the original subject)

**Validation:**
Check it in lab

### CODE REVIEW

**Review by rosspa05 — APPROVED** [review]
Tested, reviewed, and discussed. Looks good.

**Comment by rosspa05** [comment]
Discussed, tested, and reviewed. Looks better.

File: M365/MDO/ResendFailedMail.ps1 [hunk 1]
```diff
@@ -231,7 +231,7 @@ if ($failedMessages.count -ge 1000) {
 }
 
 if (-not $IncludeDuplicates) {
-    [array]$failedMessages = $failedMessages | Sort-Object MessageId -Unique
+    [array]$failedMessages = $failedMessages | Group-Object -Property MessageId | ForEach-Object { $_.Group | Sort-Object -Property Received | Select-Object -First 1 }
```
Comment (dpaulson45) [inline M365/MDO/ResendFailedMail.ps1:234]: How does this address empty subjects?

Comment (iserrano76) [reply M365/MDO/ResendFailedMail.ps1:234]: It looks an issue with Get-MessageTrace:
for all original messages it is able to get subject, but for some resend messages is not able.
With this we will get the oldest one, the original with that messageId

Comment (dpaulson45) [reply M365/MDO/ResendFailedMail.ps1:234]: Would it be better to include a where statement when the subject isn't null?

**Comment by dpaulson45** [comment]
Looks like the build pipeline is having an issue with checking out your branch. Not sure if this is due to the close or not.

File: M365/MDO/ResendFailedMail.ps1 (hunk 1 above)
Comment (iserrano76) [reply M365/MDO/ResendFailedMail.ps1:234]: I though that but if the original message is null, it will not work

**Comment by iserrano76** [comment]
> Looks like the build pipeline is having an issue with checking out your branch. Not sure if this is due to the close or not.
I try to run it again but it fail.
should we created it again?

File: M365/MDO/ResendFailedMail.ps1 (hunk 1 above)
Comment (dpaulson45) [reply M365/MDO/ResendFailedMail.ps1:234]: We talking about original message subject or original message MessageId

**Comment by dpaulson45** [comment]
Got the pipeline to work after the merge.

**Pull request merged**